src/composables/useRightClickContextMenu.ts

```diff
@@ -0,0 +1,176 @@
+            )}`;
+    } else {
+      contextMenuHeader.value = text;
+    }
+
+    if (inputSelection.isEmpty) {
+      getMenuItemButton("切り取り").disabled = true;
+      getMenuItemButton("コピー").disabled = true;
+    } else {
+      getMenuItemButton("切り取り").disabled = false;
+      getMenuItemButton("コピー").disabled = false;
+    }
+  };
+
+  const contextMenudata = ref<
+    [
+      MenuItemButton,
+      MenuItemButton,
+      MenuItemButton,
+      MenuItemSeparator,
+      MenuItemButton,
+    ]
+  >([
+    {
+      type: "button",
+      label: "切り取り",
+      onClick: async () => {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access
+        contextMenu.value?.hide();
+        await handleCut();
+      },
+      disableWhenUiLocked: false,
+    },
+    {
+      type: "button",
+      label: "コピー",
+      onClick: async () => {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access
+        contextMenu.value?.hide();
+        if (inputSelection) {
+          await navigator.clipboard.writeText(inputSelection.getAsString());
+        }
+      },
+      disableWhenUiLocked: false,
+    },
+    {
+      type: "button",
+      label: "貼り付け",
+      onClick: async () => {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access
+        contextMenu.value?.hide();
+        await handlePaste();
+      },
+      disableWhenUiLocked: false,
+    },
+    { type: "separator" },
+    {
+      type: "button",
+      label: "全選択",
+      onClick: async () => {
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-member-access
+        contextMenu.value?.hide();
+        qInputRef.value?.select();
+      },
+      disableWhenUiLocked: false,
+    },
+  ]);
+
+  const handleCut = async () => {
+    if (!inputSelection || inputSelection.isEmpty) return;
+
+    const text = inputSelection.getAsString();
+    const start = inputSelection.selectionStart;
+    setSurfaceOrYomiText(inputSelection.getReplacedStringTo(""));
+    await nextTick();
+    await navigator.clipboard.writeText(text);
+    inputSelection.setCursorPosition(start);
+  };
+
+  const setSurfaceOrYomiText = (text: string | number | null) => {
+    if (typeof text !== "string") throw new Error("typeof text !== 'string'");
+    inputText.value = text;
+  };
+
+  const handlePaste = async (options?: { text?: string }) => {
+    if (!inputSelection) return;
+
+    const text = options ? options.text : await navigator.clipboard.readText();
+    if (text == undefined) return;
+    const beforeLength = inputText.value.length;
+    const end = inputSelection.selectionEnd ?? 0;
+    setSurfaceOrYomiText(inputSelection.getReplacedStringTo(text));
+    await nextTick();
+    inputSelection.setCursorPosition(
+      end + inputText.value.length - beforeLength,
+    );
```

Comment:
ここの処理ちょっとややこしいですね！
[AudioCellのときは自動的に消される文字があってそれの考慮をしてるっぽみ](https://github.com/jdkfx/voicevox/blob/58b04317e90be649e79a6db3dcb21205ff8a55b3/src/components/Talk/AudioCell.vue#L367)。

一応このコンポーネントは切り出しが目的でしょうし、元のに合わせときますかぁ。
でもなんか元の実装でも要らない気がする･･･のでFIXMEコメント付けとく感じでどうでしょう！

```suggestion

    // NOTE: 自動的に削除される文字があることを念の為考慮している
    // FIXME: 考慮は要らないかも
    const beforeLength = inputText.value.length;
    const end = inputSelection.selectionEnd ?? 0;
    setSurfaceOrYomiText(inputSelection.getReplacedStringTo(text));
    await nextTick();
    inputSelection.setCursorPosition(
      end + inputText.value.length - beforeLength,
    );
```

Reply:
コメントを追加しました！